source/archivist.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2022 The Witness Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package source
+
+import (
+	"context"
+
+	"github.com/testifysec/go-witness/archivist"
+)
+
+type ArchivistSource struct {
+	client      *archivist.Client
+	seenGitoids []string
+}
+
+func NewArchvistSource(client *archivist.Client) *ArchivistSource {
```

Comment:
Factory functions in this package can be renamed `NewArchivist` instead of `NewArchivistSource` to reduce stuttering